recipes/docker_registry.rb

```diff
@@ -0,0 +1,116 @@
+begin
+  registry_ip = private_recipe_ip("hops","docker_registry")
+  registry_host = resolve_hostname(registry_ip)
+rescue
+  registry_host = "localhost"
+  Chef::Log.warn "could not find the docker registry ip!"
+end
+
+#for docker to push and pull from the registry it needs to trust hops_ca.pem
+case node['platform_family']
+when 'rhel'
+  cert_target = "/etc/pki/ca-trust/source/anchors/#{registry_host}.crt"
+  update_command = "update-ca-trust"
+when 'debian'
+  cert_target = "/usr/local/share/ca-certificates/#{registry_host}.crt"
+  update_command = "update-ca-certificates"
+end
+
+if ::File.exist?("#{cert_target}") === false && "#{registry_host}" != "local"
+  bash 'add_trust_cert' do
+    user "root"
+    code <<-EOH
+         ln -s #{node['kagent']['certs_dir']}/hops_ca.pem #{cert_target}
+         #{update_command}
+         EOH
+  end
+
+  #restart docker to take in account the new trusted certs
+  service 'docker' do
+    action [:restart]
+  end
+end
+
+#download registry image
+image_url = node['hops']['docker']['registry']['download_url']
+base_filename = File.basename(image_url)
+
+remote_file "/tmp/#{base_filename}" do
+  source image_url
+  backup false
+  action :create_if_missing
+  not_if "docker image inspect registry"
+end
+
+#import load registry image
+bash "import_image" do
+  user "root"
+  code <<-EOF
+    docker load -i /tmp/#{base_filename}
+  EOF
+  not_if "docker image inspect registry"
+end
+
+#delete registry image tar
+file "/tmp/#{base_filename}" do
+  action :delete
+  only_if { File.exist? "/tmp/#{base_filename}" }
+end
+
+#start docker registry
+bash "start_docker_registry" do
+  user "root"
+  code <<-EOF
+   docker run -d --restart=always --name registry -v #{node['kagent']['certs_dir']}:/certs -e REGISTRY_HTTP_ADDR=0.0.0.0:443 -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/elastic_host.pem -e REGISTRY_HTTP_TLS_KEY=/certs/priv.key -p #{node['hops']['docker']['registry']['port']}:443 registry
```

Comment:
Because I have no clue what I am doing and copied the code from somewhere else. If we need to generate new certificates I need someone that know to do it to help me.

Reply:
You can get help from @kouzant - as he's also changing the way we manage host certificates.

Reply:
So I discussed with @kouzant and it turn out that I didn't just copy past.
The name of the cert is misleading. It is a cert [generated by kagent](https://github.com/logicalclocks/kagent-chef/blob/3343d29127814c9cbf28075ba8194f94864d67e0/recipes/default.rb#L346) and composed of the [concatenation of two certs](https://github.com/logicalclocks/kagent-chef/blob/3343d29127814c9cbf28075ba8194f94864d67e0/providers/keys.rb#L36). As it is generated by kagent it is present on all the machines and I can use it.